sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -1822,8 +1849,8 @@ def hpat_pandas_series_head(self, n=5):
 
     _func_name = 'Method head().'
 
-    if not isinstance(self, SeriesType):
-        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+    ty_checker = TypeChecker(_func_name)
+    ty_checker.check(self, SeriesType)
```

Comment:
If you used `TypeChecker`, please use it everywhere in the overload. I mean raise exception for parameter `n` via the type checker.